tools/copy_bom/src/util.rs

```diff
@@ -48,6 +68,140 @@ pub fn calculate_file_hash(filename: &str) -> String {
+/// Currently, we only support dependent shared objects with absolute path.
+pub fn find_dependent_shared_objects(file_path: &str) -> HashSet<(String, String)> {
+    let mut shared_objects = HashSet::new();
+    // find dependencies for the input file
+    // first, we find the dynamic loader for the elf file, if we can't find the loader, return empty shared objects
+    let dynamic_loader = auto_dynamic_loader(file_path);
+    if dynamic_loader.is_none() {
+        return shared_objects;
+    }
+    let (dynamic_loader_src, dynamic_loader_dest) = dynamic_loader.unwrap();
+    shared_objects.insert((dynamic_loader_src.clone(), dynamic_loader_dest));
+    let output = output_of_dynamic_loader(&file_path, &dynamic_loader_src);
+    if let Ok(output) = output {
+        let mut objects = extract_dependencies_from_output(&file_path, output);
+        for item in objects.drain() {
+            shared_objects.insert(item);
+        }
+    }
+    shared_objects
+}
+
+/// get the output of the given dynamic loader.
+fn output_of_dynamic_loader(
+    file_path: &str,
+    dynamic_loader: &str,
+) -> Result<Output, std::io::Error> {
+    // if the file path has only filename, we need to add a "." directory
+    let file_path_buf = PathBuf::from(file_path);
+    let file_path = if file_path_buf.parent() == None {
+        PathBuf::from(".")
+            .join(&file_path_buf)
+            .to_string_lossy()
+            .to_string()
+    } else {
+        file_path_buf.to_string_lossy().to_string()
+    };
+    // return the output of the command to analyze dependencies
+    debug!("{} --list {}", dynamic_loader, file_path);
+    Command::new(dynamic_loader)
+        .arg("--list")
+        .arg(file_path)
+        .output()
+}
+
+/// This function will try to find a dynamic loader for a elf file automatically
+/// If we find the loader, we will return Some((loader_src, loader_dest)).
+/// This is because the loader_src and loader_dest may not be the same directory.
+/// If we can't find the loader, this function will return None
+fn auto_dynamic_loader(filename: &str) -> Option<(String, String)> {
+    let elf_file = match elf::File::open_path(filename) {
+        Err(_) => return None,
+        Ok(elf_file) => elf_file,
+    };
+    let interp_scan = match elf_file.get_section(".interp") {
+        None => return None,
+        Some(section) => section,
+    };
+    let interp_data = String::from_utf8_lossy(&interp_scan.data).to_string();
+    let loader_dest = interp_data.trim_end_matches("\u{0}"); // this interp_data always with a \u{0} at end
+    debug!("the loader of {} is {}.", filename, loader_dest);
+    let loader_dest_path = PathBuf::from(loader_dest);
+    let loader_file_name = loader_dest_path
+        .file_name()
+        .and_then(|s| s.to_str())
+        .unwrap();
+    // If the loader file name is glibc loader or musl loader, we will use occlum-modified loader
+    let loader_src = OCCLUM_LOADERS
```

Comment:
loader_src could be renamed to occlum_elf_loader, loader_dest could be renamed to inlined_elf_loader_path.
A better name could help the reviewer understand how the code works.

Reply:
fixed